components/sync15/src/sync.rs

```diff
@@ -115,25 +64,35 @@ pub fn synchronize_with_clients_engine(
     };
 
     if let Some(clients) = clients {
-        store.prepare_for_sync(clients)?;
+        store.prepare_for_sync(&|| clients.get_client_data())?;
     }
 
-    let collection_request = store.get_collection_request()?;
-    interruptee.err_if_interrupted()?;
-    let incoming_changes = IncomingChangeset::fetch(
-        client,
-        &mut coll_state,
-        collection.into(),
-        &collection_request,
-    )?;
-    assert_eq!(incoming_changes.timestamp, coll_state.last_modified);
+    let collection_requests = store.get_collection_requests()?;
+    // Should this be allowed?
```

Comment:
I added another potential option in a comment above, but I think this will get a bit of work done as I finish up the rest of remerge sync, and then again when multi-collection batches come thorough, so I'm not too worried about it.